### PR TITLE
Address PR comments in the refactor branch

### PR DESCRIPTION
**Pull request by baothientran**
This PR is part of the refactor series #481. This addresses the review comments for the following PR:
#516
#521 
#531 
#534

### CODE REVIEW

File: Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h [hunk 1]
```diff
@@ -152,6 +152,8 @@ class RasterMappedTo3DTile final {
    * will return whether there is a more detailed version of the
    * raster data available.
    *
+   * @param prepareRendererResources The IPrepareRendererResources used to
+   * create render resources for raster overlay
```
Comment (baothientran) [inline Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h:156]: Address https://github.com/CesiumGS/cesium-native/pull/516#discussion_r941905697

File: Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h [hunk 2]
```diff
@@ -162,6 +164,9 @@ class RasterMappedTo3DTile final {
 
   /**
    * @brief Detach the raster from the given tile.
+   * @param prepareRendererResources The IPrepareRendererResources used to
+   * detach raster overlay from the tile geometry
+   * @param tile The owner tile.
```
Comment (baothientran) [inline Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h:169]: Address https://github.com/CesiumGS/cesium-native/pull/516#discussion_r941905828

File: Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h [hunk 3]
```diff
@@ -192,6 +197,8 @@ class RasterMappedTo3DTile final {
    * become invalid as soon as another item is added to or removed from this
    * collection.
    *
+   * @param maximumScreenSpaceError The maximum screen space error that is used
+   * for the current tile
```
Comment (baothientran) [inline Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterMappedTo3DTile.h:201]: Address https://github.com/CesiumGS/cesium-native/pull/516#discussion_r941905904

File: Cesium3DTilesSelection/src/LayerJsonTerrainLoader.cpp [hunk 4]
```diff
@@ -622,6 +622,10 @@ Future<QuantizedMeshLoadResult> requestTileContent(
                 (pResponse->statusCode() < 200 ||
                  pResponse->statusCode() >= 300)) {
               QuantizedMeshLoadResult result;
+              result.errors.emplace_error(fmt::format(
+                  "Receive status code {} for tile content {}",
+                  pResponse->statusCode(),
+                  pRequest->url()));
```
Comment (baothientran) [inline Cesium3DTilesSelection/src/LayerJsonTerrainLoader.cpp:628]: Address https://github.com/CesiumGS/cesium-native/pull/516#discussion_r941910625

File: Cesium3DTilesSelection/src/Tileset.cpp [hunk 5]
```diff
@@ -151,7 +152,9 @@ Tileset::Tileset(
         .thenInMainThread(
             [this](
                 TilesetContentLoaderResult<CesiumIonTilesetLoader>&& result) {
-              this->_propagateTilesetContentLoaderResult(std::move(result));
+              this->_propagateTilesetContentLoaderResult(
+                  TilesetLoadType::CesiumIon,
+                  std::move(result));
```
Comment (baothientran) [inline Cesium3DTilesSelection/src/Tileset.cpp:157]: Address https://github.com/CesiumGS/cesium-native/pull/521#discussion_r941955502

File: Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterOverlayCollection.h [hunk 6]
```diff
@@ -1,8 +1,9 @@
 #pragma once
 
-#include <Cesium3DTilesSelection/Library.h>
-#include <Cesium3DTilesSelection/RasterOverlay.h>
-#include <Cesium3DTilesSelection/RasterOverlayTileProvider.h>
+#include "Library.h"
+#include "RasterOverlay.h"
+#include "RasterOverlayTileProvider.h"
+
```
Comment (baothientran) [inline Cesium3DTilesSelection/include/Cesium3DTilesSelection/RasterOverlayCollection.h:6]: Address https://github.com/CesiumGS/cesium-native/pull/531#discussion_r941967330

File: Cesium3DTilesSelection/src/RasterOverlayCollection.cpp [hunk 7]
```diff
@@ -1,7 +1,5 @@
 #include "Cesium3DTilesSelection/RasterOverlayCollection.h"
 
-#include "Cesium3DTilesSelection/Tileset.h"
-
 #include <CesiumUtility/Tracing.h>
```
Comment (baothientran) [inline Cesium3DTilesSelection/src/RasterOverlayCollection.cpp:3]: Address https://github.com/CesiumGS/cesium-native/pull/531#discussion_r941968619

File: Cesium3DTilesSelection/src/TilesetContentManager.cpp [hunk 8]
```diff
@@ -327,23 +327,20 @@ void calcRasterOverlayDetailsInWorkerThread(
...
+    auto removedProjectionIt = std::remove_if(
+        projections.begin(),
+        projections.end(),
+        [&](const CesiumGeospatial::Projection& proj) {
+          return std::find(
+                     existingProjections.begin(),
+                     existingProjections.end(),
+                     proj) != existingProjections.end();
+        });
+    projections.erase(removedProjectionIt, projections.end());
```
Comment (baothientran) [inline Cesium3DTilesSelection/src/TilesetContentManager.cpp:343]: Address https://github.com/CesiumGS/cesium-native/pull/534#discussion_r941350888 along with unit test

File: Cesium3DTilesSelection/src/LayerJsonTerrainLoader.cpp [hunk 9]
```diff
@@ -903,7 +907,7 @@ LayerJsonTerrainLoader::getLayers() const noexcept {
   return this->_layers;
 }
 
-bool LayerJsonTerrainLoader::doesTileHasUpsampledChild(const Tile& tile) const {
+bool LayerJsonTerrainLoader::tileHasUpsampledChild(const Tile& tile) const {
```
Comment (baothientran) [inline Cesium3DTilesSelection/src/LayerJsonTerrainLoader.cpp:910]: Address https://github.com/CesiumGS/cesium-native/pull/534#discussion_r941971187

**Comment by kring** [comment]
Thanks @baothientran!

**Pull request merged**